suite-native/module-send/src/components/CoinControl/SwitchCoinControlButton.tsx

```diff
@@ -0,0 +1,67 @@
+import { useNavigation } from '@react-navigation/native';
+
+import { Box, Button, Text } from '@suite-native/atoms';
+import { Icon } from '@suite-native/icons';
+import { Translation } from '@suite-native/intl';
+import {
+    RootStackParamList,
+    SendStackParamList,
+    SendStackRoutes,
+    StackToStackCompositeNavigationProps,
+} from '@suite-native/navigation';
+
+import { useUtxoSelection } from '../../hooks/useUxtoSelection';
+
+type NavigationProps = StackToStackCompositeNavigationProps<
+    SendStackParamList,
+    SendStackRoutes.SendUtxo,
+    RootStackParamList
+>;
+
+type SwitchCoinControlButtonProps = {
+    accountKey: string;
+    amount?: string;
+};
+
+export const SwitchCoinControlButton = ({ accountKey, amount }: SwitchCoinControlButtonProps) => {
+    const { selectedUtxos, totalSelectedAmount, isCoinControlEnabled } = useUtxoSelection();
+    const navigation = useNavigation<NavigationProps>();
+
+    const isMissingUtxos =
+        isCoinControlEnabled && amount && Number(totalSelectedAmount) < Number(amount);
+
+    const openCoinControlScreen = () => {
+        navigation.navigate(SendStackRoutes.SendUtxo, {
+            accountKey,
+            amount: amount ?? undefined,
+        });
+    };
+
+    let colorScheme: 'yellowBold' | 'primary' | 'tertiaryElevation0' = 'yellowBold';
+    if (isCoinControlEnabled) {
+        colorScheme = isMissingUtxos ? 'yellowBold' : 'primary';
+    } else {
+        colorScheme = 'tertiaryElevation0';
+    }
+
+    const textColor = isCoinControlEnabled && !isMissingUtxos ? 'textOnPrimary' : 'textOnTertiary';
+
+    return (
+        <Button
+            testID="switch-coin-control-button"
+            onPress={openCoinControlScreen}
+            flex={0}
+            colorScheme={colorScheme}
+        >
+            <Box flexDirection="row" alignItems="center">
+                <Icon size="mediumLarge" color={textColor} name="coins" />
+                <Box marginLeft="sp8">
+                    <Text color={textColor}>
+                        <Translation id="moduleSend.coinControl.cta" />
+                        {isCoinControlEnabled ? ` • ${selectedUtxos.length}` : ''}
+                    </Text>
+                </Box>
+            </Box>
```

Comment:
nit: use HStack for horizontal spacing instead of `marginLeft=sp8`
```suggestion
            <HStack spacing="sp8" alignItems="center">
                <Icon size="mediumLarge" color={textColor} name="coins" />
                <Text color={textColor}>
                    <Translation id="moduleSend.coinControl.cta" />
                    {isCoinControlEnabled ? ` • ${selectedUtxos.length}` : ''}
                </Text>
            </Hstack>
```

Reply:
ca79ab1df38447a6c4cfe999a927ae41a793df59